raft/src/test/java/org/apache/kafka/raft/internals/KafkaRaftMetricsTest.java

```diff
@@ -73,23 +88,35 @@ private QuorumState buildQuorumState(Set<Integer> voters) {
         );
     }
 
-    @Test
-    public void shouldRecordVoterQuorumState() {
-        QuorumState state = buildQuorumState(Utils.mkSet(localId, 1, 2));
+    @ParameterizedTest
+    @ValueSource(shorts = {0, 1})
+    public void shouldRecordVoterQuorumState(short kraftVersion) {
+        boolean withDirectoryId = kraftVersion > 0;
+        Map<Integer, VoterSet.VoterNode> voterMap = VoterSetTest.voterMap(Utils.mkSet(1, 2), withDirectoryId);
+        voterMap.put(localId, VoterSetTest.voterNode(ReplicaKey.of(localId, Optional.of(localDirectoryId))));
```

Comment:
Should we use an empty directory id if kraftVersion is 0?

Reply:
Yes. Fixed. Excuse the minor errors.